include/handler.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "types.hpp"
+
+#include <nlohmann/json.hpp>
+#include <variant>
+#include <string_view>
+
+namespace vpd
+{
+/**
+ * @brief A class to handle VPD data.
+ *
+ * The class exposes API to get parsed VPD data map, process FRUs VPD by calling
+ * respective parser, arranging them under specific interfaces and publish those
+ * VPD data over D-Bus.
+ *
+ * The class may also implement helper functions required for VPD handling.
+ */
+class Handler
+{
+  public:
+    /**
+     * List of deleted functions.
+     */
+    Handler(const Handler&);
+    Handler& operator=(const Handler&);
+    Handler(Handler&&) = delete;
+
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] json - Parsed inventory JSON.
+     */
+    Handler(const nlohmann::json& json) : m_inventoryJson(json)
+    {
+    }
+
+    /**
+     * @brief Destructor
+     */
+    ~Handler() = default;
+
+    /**
+     * @brief An API to process VPD.
+     *
+     * This api is exposed to process VPD map recieved as a parameter and call
+     * respective API to populate VPD under interfaces and publish them on
+     * D-Bus.
+     *
+     * @param[in] vpdFilePath - EEPROM path of the FRU whose data needs to be
+     * handled.
+     * @param[in] vpdMap - Map of parsed VPD data.
```

Comment:
Doxygen is missing for ParsedVPD parameter, please add it.

Reply:
ParsedVPD is not a parameter. Doxygen is not required for that.